actors/builtin/miner/miner_actor.go

```diff
@@ -416,22 +418,33 @@ func (a Actor) SubmitWindowedPoSt(rt Runtime, params *SubmitWindowedPoStParams)
 // Proposals must be posted on chain via sma.PublishStorageDeals before PreCommitSector.
 // Optimization: PreCommitSector could contain a list of deals that are not published yet.
 func (a Actor) PreCommitSector(rt Runtime, params *SectorPreCommitInfo) *adt.EmptyValue {
```

Comment:
Most of these checks were already here, I just put them in field order,and switched to RequireParam. Please check _carefully_.